src/garage/experiment/local_runner.py

```diff
@@ -60,7 +62,11 @@ def __init__(self, snapshot_config, max_cpus=1):
         self.plot = False
 
         self._setup_args = None
-        self.train_args = None
+        self._train_args = None
+        self._stats = types.SimpleNamespace(total_itr=0,
```

Comment:
is there a reason to keep using `SimplerNamespace` over `namedtuple`? I think `SimpleNamespace` messes with type inference.

Reply:
...and why put these into their own type anyway? can they just be fields or a dict?

i see why you want to group them, and why not to use `namedtuple` (because it's immutable). perhaps you could just define real classes for these though. it's not much boilerplate and it would make it easier to see/validate exactly what you expect to be in these variables.

i think as they get bigger, these types are starting to hide some complexity/assumptions.

Reply:
They are immutable in terms of fields. The values are actually mutable. For example, `stats` gets updated every epoch. 

Originally I used dict for `setup_args` and `train_args`, but you suggested `SimpleNamespace` for better visibility. (in #633 , [comment](https://github.com/rlworkgroup/garage/pull/633#discussion_r277916964))

I think both `nametuple` and `SimpleNamespace` are better than dict because they prescribe fields. Where do you think I should place these types? My first idea is to put them out of `LocalRunner` but still keep them in this file.

Reply:
Yes I agree. It hasn't worked out -- what do you think?

I think that namedtuple, dict, and SimpleNamespace don't fit the bill. Since they are becoming key types for the Runner, I am suggesting we make them (simple) real classes.

Reply:
So far these args and stats are just key-value pairs, with a fixed set of fields. No validation, no transformation, so I think namedtuple is sufficient.

Reply:
alright, please give namedtuple a try

Reply:
Oh, I totally missed that. Yes, they need to be mutable. Okay, I will make them into classes.